### PR TITLE
[enhancement](bitmap)support bitmap type for non-key  column in unique table

### PR DESCRIPTION
**Pull request by bobhan1**
## Proposed changes

add support for bitmap type for non-key column in unique table
## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by bobhan1** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 46.12 seconds
 stream load tsv:          539 seconds loaded 74807831229 Bytes, about 132 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          29.5 seconds inserted 10000000 Rows, about 338K ops/s
 storage size: 17162072800 Bytes

**Comment by bobhan1** [comment]
run buildall

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 45.76 seconds
 stream load tsv:          556 seconds loaded 74807831229 Bytes, about 128 MB/s
 stream load json:         20 seconds loaded 2358488459 Bytes, about 112 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          30 seconds loaded 861443392 Bytes, about 27 MB/s
 insert into select:          29.3 seconds inserted 10000000 Rows, about 341K ops/s
 storage size: 17162193388 Bytes

**Comment by bobhan1** [comment]
run feut

**Comment by bobhan1** [comment]
run beut

**Review by zhannngchen — APPROVED** [review]
LGTM

**Comment by BiteTheDDDDt** [comment]
Why support unique table to store bitmap columns? This seems a bit inefficient

File: fe/fe-core/src/main/java/org/apache/doris/analysis/CreateTableStmt.java [hunk 1]
```diff
@@ -519,19 +517,23 @@ public void analyze(Analyzer analyzer) throws UserException, AnalysisException {
...
+                    if (keysDesc.getKeysType() == KeysType.DUP_KEYS) {
+                        throw new AnalysisException("column:" + columnDef.getName()
+                                + " must be used in AGG_KEYS or UNIQUE_KEYS.");
+                    }
+                } else {
+                    if (keysDesc.getKeysType() != KeysType.AGG_KEYS) {
+                        throw new AnalysisException("column:" + columnDef.getName() + " must be used in AGG_KEYS.");
+                    }
+                }
             }
```
Comment (BiteTheDDDDt) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/CreateTableStmt.java:530]: This pr also changed the behavior of hll and quantile_state, should make corresponding changes to their documents, and add case

Comment (bobhan1) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/CreateTableStmt.java:530]: > This pr also changed the behavior of hll and quantile_state, should make corresponding changes to their documents, and add case

the behavior of hll and quantile_state is not changed, only bitmap type is allowed to appear in unique table.

**Review by BiteTheDDDDt — APPROVED** [review]

**Comment by zhannngchen** [comment]
run clickbench-new

**Comment by hello-stephen** [comment]
(From new machine)TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 48.95 seconds
 stream load tsv:          556 seconds loaded 74807831229 Bytes, about 128 MB/s
 stream load json:         21 seconds loaded 2358488459 Bytes, about 107 MB/s
 stream load orc:          65 seconds loaded 1101869774 Bytes, about 16 MB/s
 stream load parquet:          32 seconds loaded 861443392 Bytes, about 25 MB/s
 insert into select:          30.6 seconds inserted 10000000 Rows, about 326K ops/s
 storage size: 17161990440 Bytes

**Pull request merged**